### PR TITLE
v4.0 changelog

### PR DESCRIPTION
**Pull request by jasonblais**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,71 +1,121 @@
...
+ [#512](https://github.com/mattermost/desktop/issues/512)
  - Added the option to bounce the Dock icon when receiving new messages.
  [#514](https://github.com/mattermost/desktop/issues/514)
 
+### Architectural Changes
+ - Major version upgrade of Electron from v1.6.11 to v1.7.9. Electron is the underlying technology used to build the Desktop apps.
+ [#602](https://github.com/mattermost/desktop/pull/602)
+ - The app now uses CSS to style the user interface. Styles are also divided to React's inline `style` and CSS.
+ [#540](https://github.com/mattermost/desktop/pull/540)
+ - Yarn is now used to manage dependancies across Windows, Mac and Linux builds.
```
Comment (MusikPolice) [inline CHANGELOG.md]: Is that supposed to be "dependencies"?

File: CHANGELOG.md [hunk 2]
```diff
@@ -1,71 +1,121 @@
...
+ - Fixed a mis-aligned `+` button in the server tab bar.
+ [#541](https://github.com/mattermost/desktop/issues/541)
 
 #### Linux
- - Fixed the main window not minimized when the app is launched via "Start app on Login" option.
+ - Fixed the main window not being minimized when the app is launched via "Start app on Login" option.
  [#570](https://github.com/mattermost/desktop/issues/570)
 
+### Known Issues
+
```
Comment (MusikPolice) [inline CHANGELOG.md]: Should we put "None" or "N/A" under this heading just so it isn't floating around in space?

Comment (jasonblais) [reply CHANGELOG.md]: There are known issues for All Platforms, and Windows/Mac/Linux separately. So I think it's appropriate to have this heading

**Comment by jasonblais** [comment]
@yuya-oc ready for review

File: CHANGELOG.md [hunk 3]
```diff
@@ -1,71 +1,120 @@
...
 
-## UNDER DEVELOPMENT
+## Release v4.0.0
 
-The "UNDER DEVELOPMENT" section of the Mattermost Desktop changelog appears
-in the product's `master` branch to note key changes committed to `master`
-and are on their way to the next stable release.
-When a stable release is pushed, "UNDER DEVELOPMENT" heading is removed
-from the final changelog of the release.
+Release date: December, 2017
```
Comment (MusikPolice) [inline CHANGELOG.md]: This should be changed to "January, 2018", shouldn't it?

Comment (jasonblais) [reply CHANGELOG.md]: Good catch, updated the date

File: CHANGELOG.md [hunk 4]
```diff
@@ -1,71 +1,120 @@
...
+### Architectural Changes
+ - Major version upgrade of Electron from v1.6.11 to v1.7.9. Electron is the underlying technology used to build the Desktop apps.
+ [#602](https://github.com/mattermost/desktop/pull/602)
+ - The app now employs Mattermost APIv4 endpoints.
+ - The app now uses CSS to style the user interface. Styles are also divided to React's inline `style` and CSS.
+ [#540](https://github.com/mattermost/desktop/pull/540)
+ - Yarn is now used to manage dependencies across Windows, Mac and Linux builds.
+ [#485](https://github.com/mattermost/desktop/issues/485)
+ - Build is now run automatically before packaging the apps with `npm run package`.
+ [#590](https://github.com/mattermost/desktop/pull/590)
```
Comment (MusikPolice) [inline CHANGELOG.md]: Issues #485 and #590 might be of interest to developers, but don't really impact users. Are they appropriate for inclusion in release notes? If our policy is to include all issues, that's fine. Just wondering what the intended target audience is?

**Comment by MusikPolice** [comment]
@jasonblais LGTM. Feel free to merge once @yuya-oc has approved.

File: CHANGELOG.md [hunk 5]
```diff
@@ -1,71 +1,120 @@
...
 
 #### Mac
- - Added `Ctrl+Tab` and `Ctrl+Shift+Tab` shortcuts to switch tabs
- [#512](https://github.com/mattermost/desktop/issues/512)
- - Added the feature to open the application via `mattermost://` link.
+ - Added support for protocol deep linking where the desktop app opens via `mattermost://` link if app is already installed.
  [#616](https://github.com/mattermost/desktop/pull/616)
+ - Added `Ctrl+Tab` and `Ctrl+Shift+Tab` shortcuts to switch between server tabs,
+ [#512](https://github.com/mattermost/desktop/issues/512)
  - Added the option to bounce the Dock icon when receiving new messages.
```
Comment (yuya-oc) [inline CHANGELOG.md]: `messages` -> `notifications` due to #683.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,71 +1,120 @@
...
  [#514](https://github.com/mattermost/desktop/issues/514)
 
+### Architectural Changes
+ - Major version upgrade of Electron from v1.6.11 to v1.7.9. Electron is the underlying technology used to build the Desktop apps.
+ [#602](https://github.com/mattermost/desktop/pull/602)
+ - The app now employs Mattermost APIv4 endpoints.
```
Comment (yuya-oc) [inline CHANGELOG.md]: I think that API is not used yet. It will be used for #608 to detect whether the user is logged in.

**Comment by jasonblais** [comment]
@yuya-oc Updated, good points.

**Comment by yuya-oc** [comment]
Thanks all!

**Pull request merged**